### PR TITLE
Make all github calls as receivable by github client

### PR DESCRIPTION
**Pull request by srinivashegde86**
All the methods are direct function calls that can be made directly without initializing the github client. Updating this so that:

1. We can get a valid github client to perform github actions
1. Add the methods to the client so that we can call client.<Method> instead of relying on a global var
1. We can get a valid github user to perform actions on it directly
1. Update github_issue to hold these values and make calls to the issue struct directly

### CODE REVIEW

**Comment by srinivashegde86** [comment]
will wait for https://github.com/knative/test-infra/pull/506 to be merged as there are some conflicts.

/hold

**Comment by srinivashegde86** [comment]
/hold cancel
the PR has now merged.

File: tools/flaky-test-reporter/ghutil/ghutil.go [hunk 1]
```diff
@@ -40,56 +40,65 @@ const (
...
+	*github.Client
+}
+
+// GithubUser provides methods to perform operations as a github user
+type GithubUser struct {
+	*github.User
+}
+
+// GetGithubClient explicitly authenticates to github with giving token and returns a handle
+func GetGithubClient(tokenFilePath string) (*GithubClient, error) {
```
Comment (adrcunha) [inline tools/flaky-test-reporter/ghutil/ghutil.go]: nit: shouldn't this be called NewGithubClient then?

**Pull request merged**